### DISCUSSION TITLE
PrometheusNotIngestingSamples rule alerting for type "histogram"

### DISCUSSION THREAD

**Discussion by frederiksf** [prometheus-operator/kube-prometheus#2095] [General]
Since we move from version 0.10 to version 0.12 (prometheus release 2.41.0) the following alerting rule fires

name: [PrometheusNotIngestingSamples](https://prometheus.dev.net/graph?g0.expr=ALERTS%7Balertname%3D%22PrometheusNotIngestingSamples%22%7D&g0.tab=1&g0.stacked=0&g0.show_exemplars=0.g0.range_input=1h.)
expr: [rate(prometheus_tsdb_head_samples_appended_total{job="prometheus-k8s",namespace="monitoring"}[5m]) <= 0](https://prometheus.dev.net/graph?g0.expr=rate(prometheus_tsdb_head_samples_appended_total%7Bjob%3D%22prometheus-k8s%22%2Cnamespace%3D%22monitoring%22%7D%5B5m%5D)%20%3C%3D%200&g0.tab=1&g0.stacked=0&g0.show_exemplars=0.g0.range_input=1h.)
for: 10m
labels:
severity: warning
annotations:
description: Prometheus {{$labels.namespace}}/{{$labels.pod}} is not ingesting samples.
summary: Prometheus is not ingesting samples.

After some research i see that type floating has values > 0 but type histogram is the one which fires the alarm. 

![image](https://user-images.githubusercontent.com/3944511/235902893-11488711-87b9-4f1a-aaa8-9780bd53a34e.png)

What i do not understand is that i have histogram metrics which i can scrape successfull. for example:
![image](https://user-images.githubusercontent.com/3944511/235904335-5ab64caa-e6ef-48a0-9eff-daa806ad5533.png)

In the previous version of prometheus (2.32.1) the was only one metric without the difference between type=count or type=histogram

![image](https://user-images.githubusercontent.com/3944511/235903643-1401b57c-12b1-4cc8-955d-be6ea70856af.png)

Does anybody understand what is happening here? Thank you!

**Comment by VerosK** [comment]
Wasn't this issue resolved in https://github.com/prometheus-operator/kube-prometheus/discussions/2095 ?

**Reply by palmobar** [reply]
@VerosK this is the issue #2095 x| @frederiksf did you manage to resolve it?

**Comment by tamsky** [comment]
Author opened both a discussion (#2095) and an issue (#2096).

Issue appears unresolved, I'm seeing this behavior as well.

Code was added here to create separate {type="float|histogram"} labelled metrics:
- https://github.com/prometheus/prometheus/commit/d0a6488c74b7f40de5e50873eb8452799f260455#diff-c3ae701c35bb1de34870207029d1fb3390613fbba65e6f2f9bffe059bff89001R765-R766

Current rule @current{main} ::
- https://github.com/prometheus-operator/kube-prometheus/blob/858a2a53f42720a8baa048de5984835deb59328d/manifests/prometheus-prometheusRule.yaml#L111

This rule probably should be updated to alert only on the old behavior by adding the `type="float"` to the labelset.
